### DISCUSSION TITLE
vlc: cannot start program (snap)

### DISCUSSION THREAD

**Discussion by LucasFelipelli** [netblue30/firejail#4651] [Q&A] [answered]
Running `firejail vlc` shows the following error:

> Error: no suitable vlc executable found

**Comment by rusty-snake** [comment]
Is vlc installed? How did you installed firejail/vlc? In which distro? In which version? Does `vlc` work? Does `firejail --noprofile vlc` work? Do you have any modifications (e.g. globals.local)?

**Comment by LucasFelipelli** [comment]
1. Yes
2. `sudo apt install firejail/vlc`
3. Ubuntu
4. 20.04
5. Yes
6. No
7. No

```console
$ firejail --noprofile vlc
Parent pid 37544, child pid 37545
Child process initialized in 7.86 ms
2021/10/27 17:54:53.650274 system_key.go:154: cannot determine cgroup version: cannot determine filesystem type: cannot statfs path: permission denied
internal error, please report: running "vlc" failed: cannot create transient scope: DBus error "org.freedesktop.DBus.Error.InvalidArgs": [Process 2 is a kernel thread, refusing.]

Parent is shutting down, bye...
```

**Reply by rusty-snake** [reply]
4. I meant the firejail version
6. Bad. Is `vlc` the only program broken in firejail? Or is even something simple like `firejail --noprofile true` affected?

**Reply by LucasFelipelli** [reply]
firejail version 0.9.62
As of now, only vlc

**Reply by rusty-snake** [reply]
Can you try with the [noprofile](https://gist.github.com/rusty-snake/bb234cb3e50e1e4e7429f29a7931cc72).

**Comment by netblue30** [comment]
Added in noprofile.profile in git!

**Comment by kmk3** [comment, marked as answer]
@Archangelll [on Oct 27, 2021](https://github.com/netblue30/firejail/discussions/4651#discussioncomment-1547753):

> ```
> $ firejail --noprofile vlc
> Parent pid 37544, child pid 37545
> Child process initialized in 7.86 ms
> 2021/10/27 17:54:53.650274 system_key.go:154: cannot determine cgroup version: cannot determine filesystem type: cannot statfs path: permission denied
> ```

vlc is not written in Go, and "system_key.go" seems to be a snap thing.

So presumably vlc was installed as a snap, which is not supported by firejail.

Solution: Uninstall the snap and install the normal .deb package (or use an
AppImage).
